Dockerfile

```diff
@@ -0,0 +1,83 @@
+FROM ubuntu:23.04 as cogip-console
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get install -y \
+        python3 \
+        python3-pip \
+        python3-venv \
+        git \
+        cmake \
+        swig \
+        socat \
+        wait-for-it
+
+RUN python3 -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+RUN python -m pip install -U pip setuptools wheel
+
+ADD requirements.txt requirements-dev.txt pyproject.toml /src/
+RUN python -m pip install -r /src/requirements.txt -r /src/requirements-dev.txt
+
+ADD cogip /src/cogip
+RUN python -m pip install -e /src[dev]
+
+CMD ["sleep", "infinity"]
+
+
+FROM cogip-console as cogip-gui
+
+RUN apt-get install -y \
+        libgl1 \
+        libegl1 \
+        libglib2.0-0 \
+        libxkbcommon0 \
+        libdbus-1-3 \
+        libnss3 \
+        libxcomposite1 \
+        libxdamage1 \
+        libxrender1 \
+        libxrandr2 \
+        libxtst6 \
+        libxi6 \
+        libasound2 \
+        libxkbfile1 \
+        libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1 \
+        libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0
+#bsdextrautils groff-base libice6 libpipeline1 libsm6 libuchardet0 libxaw7 libxcb-damage0 libxcursor1 libxft2 libxmu6 libxt6 man-db xbitmaps
+
+FROM ubuntu:20.04 as cogip-firmware
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
```

Comment:
you forgot to do `apt-get update` here.

Should be:
```Dockerfile
RUN apt-get update && \                                                         
    apt-get install -y \
```

This leads to error like this:
```
6.399 Get:71 http://archive.ubuntu.com/ubuntu lunar/main amd64 libxcb-render-util0 amd64 0.3.9-1build3 [10.3 kB]
6.410 E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/a/alsa-ucm-conf/alsa-ucm-conf_1.2.6.3-1ubuntu9.1_all.deb  404  Not Found [IP: 91.189.91.82 80]
6.410 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/libx/libx11/libx11-xcb1_1.8.4-2ubuntu0.2_amd64.deb  404  Not Found [IP: 91.189.91.82 80]
6.410 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
6.410 Fetched 44.4 MB in 5s (9657 kB/s)
------
Dockerfile:32
--------------------
  31 |     
  32 | >>> RUN apt-get install -y \
  33 | >>>         libgl1 \
  34 | >>>         libegl1 \
  35 | >>>         libglib2.0-0 \
  36 | >>>         libxkbcommon0 \
  37 | >>>         libdbus-1-3 \
  38 | >>>         libnss3 \
  39 | >>>         libxcomposite1 \
  40 | >>>         libxdamage1 \
  41 | >>>         libxrender1 \
  42 | >>>         libxrandr2 \
  43 | >>>         libxtst6 \
  44 | >>>         libxi6 \
  45 | >>>         libasound2 \
  46 | >>>         libxkbfile1 \
  47 | >>>         libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1 \
  48 | >>>         libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0
  49 |     #bsdextrautils groff-base libice6 libpipeline1 libsm6 libuchardet0 libxaw7 libxcb-damage0 libxcursor1 libxft2 libxmu6 libxt6 man-db xbitmaps
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get install -y         libgl1         libegl1         libglib2.0-0         libxkbcommon0         libdbus-1-3         libnss3         libxcomposite1         libxdamage1         libxrender1         libxrandr2         libxtst6         libxi6         libasound2         libxkbfile1         libxcb-xkb1 libxcb-image0 libxcb-render-util0 libxcb-render0 libxcb-util1         libxcb-icccm4 libxcb-keysyms1 libxcb-shape0 libxkbcommon-x11-0" did not complete successfully: exit code: 100
ERROR: Service 'monitor' failed to build : Build failed
```

Reply:
I did not forget, it's just a typo ;-)
Good catch!
Done